.github/workflows/test-deploy.yml

```diff
@@ -176,11 +185,37 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.2'
+          tools: composer:v2
+
+      - name: Get Composer cache directory
+        id: composer-cache
+        run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+      - name: Cache Composer packages
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('composer.lock') }}
+          restore-keys: ${{ runner.os }}-composer-
+
       - name: Setup Node.js
         uses: actions/setup-node@v4
         with:
           node-version-file: '.nvmrc'
-          cache: 'npm'
+
+      - name: Cache npm dependencies
+        uses: actions/cache@v4
+        with:
+          path: |
+            node_modules
+            themes/*/node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('package-lock.json', '**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
```

Comment:
Best not to provide a restore-key fallback, as we should run an npm clean-install nonetheless if we have changes to the package lock. Leave just the "key:"

Reply:
Interesting, this was caused by a node module that cared for the CPU type 
_Valid cpu:  arm
npm error notsup Actual cpu: x64_

Maybe try with `npm install --cache .npm --prefer-offline --no-audit` instead ?
See if it is any faster on cache fails.